Include/internal/pycore_regions.h

```diff
@@ -14,6 +14,21 @@ extern "C" {
 #define Py_CHECKWRITE(op) ((op) && !_Py_IsImmutable(op))
 #define Py_REQUIREWRITE(op, msg) {if (Py_CHECKWRITE(op)) { _PyObject_ASSERT_FAILED_MSG(op, msg); }}
 
+void _Py_SET_TAGGED_REGION(PyObject *ob, Py_region_ptr_with_tags_t region);
+#if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 < 0x030b0000
+#  define Py_SET_TAGGED_REGION(ob, region) _Py_SET_TAGGED_REGION(_PyObject_CAST(ob), (region))
+#endif
+
+static inline void _Py_SET_REGION(PyObject *ob, Py_region_ptr_t region) {
+    _Py_SET_TAGGED_REGION(ob, Py_region_ptr_with_tags(region & Py_REGION_MASK));
+}
+#if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 < 0x030b0000
+#  define Py_SET_REGION(ob, region) (_Py_SET_REGION(_PyObject_CAST(ob), _Py_CAST(Py_region_ptr_t, (region))))
+#endif
+
+int _Py_IsLocal(PyObject *op);
+#define Py_IsLocal(op) _Py_IsLocal(_PyObject_CAST(op))
```

Comment:
The macro name is missing a leading _ to be consistent with _Py_IsImmutable etc.

Reply:
Using `_Py_IsLocal` for the macro breaks compilation. It seems to be the function needs to be in the same file as the macro to allow this kind of overwriting. I tried `__Py_IsLocal` but that triggers an "smelly function name" lint -.-